lib/prompt.js

```diff
@@ -5,14 +5,25 @@ const { promisify } = require('util');
 const read = promisify(require('read'));
 
 async function prompt(message, options) {
+    let value;
+
     // Read input
-    let value = await read({
-        prompt: message,
-        silent: options.silent,
-        replace: options.replace,
-        input: options.input,
-        output: options.output,
-    });
+    // Manage timeout
+    try {
+        value = await read({
+            prompt: message,
+            silent: options.silent,
+            replace: options.replace,
+            input: options.input,
+            output: options.output,
+            timeout: options.timeout,
+        });
+    } catch (err) {
+        if (err.message !== 'timed out' || !options.default) {
```

Comment:
Cant we use err.code instead?

Reply:
err.code seems undefined to me

Reply:
That sucks. I would prefer to add a code property, with a TIMEDOUT value.

Reply:
What do you mean? Create a const TIMEDOUT = "timed out" ? Or considering patch the read package (which would be too slow)

Reply:
Sorry, I meant that we could add a .code property to the error thrown, and rethrow it. This way it would be easier to test for a timeout error for users using this package.

Reply:
Bump

Reply:
I do not really like to modify exception from lower library to rethrow them. The code will have to be export as const also to be truly useful, so sorry to ask but can you add the exact implementation you would like or update the PR (i think you have permissions to)

Reply:
Well maybe then we can create a new error, with the code and throw it?

```js
throw Object.assign(new Error(err.message), { code: ‘TIMEDOUT’ })
```

Reply:
Just a side-note. This is to improve the DX if you need to check if the error is a timeout. Checking messages is flaky, that’s why all nodejs errors have a code.

Reply:
Yes I understand the interest of having a more clean Error, I think node is not enforcing that well enough, as Error constructor only takes message.

I'll update with your line

Reply:
Done